java/src/test/java/org/wildfly/openssl/BasicOpenSSLEngineTest.java

```diff
@@ -158,6 +158,14 @@ public void testSingleEnabledProtocolTLS13() throws IOException, InterruptedExce
         testSingleEnabledProtocolBase("TLSv1.3");
     }
 
+    @Test
+    public void tesRDRANDEngine() throws IOException, InterruptedException {
+        System.setProperty(SSL.ORG_WILDFLY_OPENSSL_ENGINE, "rdrand");
+        SSL ssl = SSL.getInstance();
```

Comment:
Just tried running this test but noticed that the configured custom engine doesn't actually get used since `SSL#getInstance()` is already called during the test setup so `SSL#init` doesn't get run again.

To make sure that `SSL#getInstance` actually makes use of the configured property, I think we'd need to create a new test class with this test case and update the `maven-surefire-plugin` configuration in the[ java/pom.xml](https://github.com/wildfly-security/wildfly-openssl/blob/main/java/pom.xml#L119-L136) file so that it sets the `org.wildfly.openssl.engine` property to `rdrand` for this new test class only.

As an example, we updated the `maven-surefire-plugin` configuration for a different testsuite recently, where we needed to specify some properties using the -D option when a certain test was being run:

https://github.com/wildfly/wildfly/pull/13673/commits/d23461e57917395f730b8fd84ca5a8f4da322d64#diff-16653cf23553258b76452eb02be79fc3d93668abe53bd164f0f3dd10a09a9fe7R292-R323

I think we should be able to do something similar here.

Reply:
okay. I will take a look at this and try to refine the testcase.

Reply:
I have modified the code in wildfly-openssl-native, please review the code https://github.com/wildfly-security/wildfly-openssl-natives/pull/16/commits/e87919186332bd57d9537153f18de37c1f2bccb0

The main change is to throw exceptions if any error occurs during initializing the custom engine.

So I also add the testcase to verify this function.
I think adding a new testcase class is simple than using maven-surefire-plugin method.

Reply:
@heyuanliu-intel Great! Thanks very much for the updates here and on the wildfly-openssl-natives PR. We will take a look this week. Thanks!